File: scripts/update-ui-imports.mjs
#!/usr/bin/env node
/**
 * Script to update import paths in UI components after extraction.
 * Updates:
 * - @/lib/utils -> ../../utils
 * - @/components/ui/ -> ./
 */

import { readFileSync, writeFileSync } from "fs";
import { readdir } from "fs/promises";
import { join, dirname } from "path";
import { fileURLToPath } from "url";

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

const UI_COMPONENTS_DIR = join(__dirname, "..", "packages", "ui", "src", "components", "ui");

async function updateImportsInFile(filePath) {
	try {
		const content = readFileSync(filePath, "utf-8");
		let updated = content;

		// Replace @/lib/utils with ../../utils
		updated = updated.replace(/from "@\/lib\/utils"/g, 'from "../../utils"');

		// Replace @/components/ui/ with ./
		updated = updated.replace(/from "@\/components\/ui\//g, 'from "./');

		if (updated !== content) {
			writeFileSync(filePath, updated, "utf-8");
			return true;
		}
		return false;
	} catch (error) {
		console.error(`Error processing ${filePath}:`, error.message);
		return false;
	}
}

async function main() {
	try {
		const files = await readdir(UI_COMPONENTS_DIR);
		const tsxFiles = files.filter((f) => f.endsWith(".tsx"));

		let updatedCount = 0;
		for (const file of tsxFiles) {
			const filePath = join(UI_COMPONENTS_DIR, file);
			if (await updateImportsInFile(filePath)) {
				updatedCount++;
				console.log(`Updated: ${file}`);
			}
		}

		console.log(`\nCompleted: Updated ${updatedCount} of ${tsxFiles.length} files`);
	} catch (error) {
		console.error("Error:", error.message);
		process.exit(1);
	}
}

main();

File: scripts/update-ui-imports.py
#!/usr/bin/env python3
"""
Script to update import paths in UI components after extraction.
Updates:
- @/lib/utils -> ../../utils
- @/components/ui/ -> ./
- @/hooks/ -> (leaves as-is for now, will need to be handled separately)
"""

import os
import re
from pathlib import Path

UI_COMPONENTS_DIR = Path(__file__).parent.parent / "packages" / "ui" / "src" / "components" / "ui"

def update_imports_in_file(file_path: Path) -> bool:
    """Update import paths in a single file."""
    try:
        with open(file_path, "r", encoding="utf-8") as f:
            content = f.read()
        
        original_content = content
        
        # Replace @/lib/utils with ../../utils
        content = re.sub(r'from "@/lib/utils"', 'from "../../utils"', content)
        
        # Replace @/components/ui/ with ./
        content = re.sub(r'from "@/components/ui/', 'from "./', content)
        
        if content != original_content:
            with open(file_path, "w", encoding="utf-8") as f:
                f.write(content)
            return True
        return False
    except Exception as e:
        print(f"Error processing {file_path}: {e}")
        return False

def main():
    """Main function to update all UI component files."""
    if not UI_COMPONENTS_DIR.exists():
        print(f"Error: Directory {UI_COMPONENTS_DIR} does not exist")
        return
    
    updated_count = 0
    total_count = 0
    
    for file_path in UI_COMPONENTS_DIR.glob("*.tsx"):
        total_count += 1
        if update_imports_in_file(file_path):
            updated_count += 1
            print(f"Updated: {file_path.name}")
    
    print(f"\nCompleted: Updated {updated_count} of {total_count} files")

if __name__ == "__main__":
    main()

